### ISSUE TITLE
Best Time to Buy and Sell Stock: Unrecognized symbol `toArray()`

### ISSUE THREAD

**Issue by iamdavidmichaelmoore** [LeetCode-Feedback/LeetCode-Feedback#13326]
#### Your LeetCode username

ID: datamathcode

#### Category of the bug
- [x] Question
- [x] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug

I think the function I wrote should work because it works in my IDE. But I get the following feedback in Leetcode.

![image](https://user-images.githubusercontent.com/31048659/236657731-e1587db2-26fc-4d5e-affc-29d061686300.png)

I noticed that when I broke apart the expression in the return statement, the `toArray()` function received an error in the Leetcode output. The function is unknown. Am I doing something wrong in my code?

```Bash
Line 19: error: cannot find symbol
        System.out.println(results.toArray[3]);
                                  ^
  symbol:   variable toArray
  location: variable results of type HashSet<Integer>
```
![image](https://user-images.githubusercontent.com/31048659/236657667-913f302c-581b-4dd0-9762-8539ae86394c.png)

#### Code you used for Submit/Run operation

```Java
public int maxProfit(int[] prices) {
     record Stock (Integer day,Integer price){}
     int size = prices.length;
     ArrayList<Stock> stocks = new ArrayList<>();
     for (int i = 0; i < size; i++) {
         stocks.add(new Stock(i+1,prices[i]));
     }

     Collections.sort(stocks, (a,b) -> a.price.compareTo(b.price));

     HashSet<Integer> results = new HashSet<>();
     for (int i = 0; i < size; i++) {
         if (stocks.get(0).day < stocks.get(i).day) {
             results.add(Math.abs(stocks.get(0).price - stocks.get(i).price));
         }
     } 
     return results.toArray(new Integer[results.size()])[results.size()-1];
 }
```

#### Language used for code

Java

#### Expected behavior

The following line of code (below) should return the last integer in the array. Further, an array of equal size is created by the `toArray()` function call on the HashSet.

```Java
return results.toArray(new Integer[results.size()])[results.size()-1];
```

#### Screenshots

See the terminal output in the screenshot.
![image](https://user-images.githubusercontent.com/31048659/236657567-871c6a4a-1442-4597-a7b1-db2b1c944507.png)

#### Additional context

I used JavaSE-20 in VSCode to run and debug the function. I located the version in Leetcode, noting that it is JavaSE-17. However, I am not sure that it makes a difference.

**Comment by LC-Pam** [comment]
Thank you for reaching out to us! Unfortunately, we are unable to troubleshoot the different behavior between your local IDE and the LeetCode environment.

For this issue, please raise this as a discuss post in the problem discuss section as other users may have run into the same issue and resolved it.

At LeetCode, we care about the whole learning experience that comes with each problem and we encourage users to teach and help each other to enrich their learning. In addition, we want our users to debug their code and learn from the process.

We appreciate your time and support.

**Comment by iamdavidmichaelmoore** [comment]
Okay. I'll do that. Thank you.

And for your situational awareness, the reason I posted my request for help
on GH is due to a link within Leetcode that indicates GH is a better
option. So your reply is somewhat deflating and confusing more broadly.
Secondly, the nature of the issue that I've posted on GH appears to be a
bug and not user error.

Again, I'll post this issue on Leetcode as instructed. It's just that in
light of your reply what led me to GH is confusing in hindsight .
